app/controllers/course/enrol_requests_controller.rb

```diff
@@ -0,0 +1,128 @@
+class Course::EnrolRequestsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+  skip_before_filter :verify_authenticity_token, only: [:approve_selected, :delete_selected]
+
+  def index #:nodoc:
+    @staff_requests = @enrol_requests.staff
+    @student_requests = @enrol_requests.student
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def new #:nodoc:
+    unless current_user
+      redirect_to new_user_session_path
+      return
+    end
+    unless @course.opened?
+      redirect_to course_path(@course), flash: {error: t('.course_not_open')}
+      return
+    end
+
+    existing = Course::EnrolRequest.find_by_user_id_and_course_id(current_user.id, @course.id)
+    if existing
+      @enrol_request = existing
+    end
+
+    if !current_course_user
+      if CourseUser.roles.include?(params[:role])
+        @enrol_request.role = params[:role]
+      else
+        @enrol_request.role = :student
+      end
+
+      @enrol_request.course = @course
+      @enrol_request.user = current_user
+      @enrol_request.save!
+      # TODO notify lecturer
```

Comment:
Annotation keywords should be all upper case, followed by a colon and a space, then a note describing the problem.